atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueue.java

```diff
@@ -190,6 +191,21 @@ public int getNumShards() {
         return numShards.get();
     }
 
+    /**
+     * Returns the most recently known number of shards for the given strategy.
+     */
+    public int getNumShards(SweeperStrategy strategy) {
```

Comment:
hmm do not love that we have this `exceptional` behavior for Strategy.NOTHING but do not see a better way for now.

Reply:
I mean, in hindsight we should have decoupled number of shards for thorough/conservative and this would have made more sense now we just live with the consequences and I do not think migrating at this point is something we should do